src/main/java/software/bernie/techarium/pipe/networks/PipeNetwork.java

```diff
@@ -127,7 +135,36 @@ public void deprecateSpecific(ServerWorld world, Map<BlockPos, UUID> newNetworkU
         }
     }
 
-    public abstract Filter<ToTransport> getFilter(PipePosition pipePosition);
+    public Filter<ToTransport> getFilter(ServerWorld world, PipePosition pipePosition, boolean input) {
+        TileEntity tileEntity = world.getBlockEntity(pipePosition.getPos());
+        if (tileEntity instanceof PipeTile) {
+            PipeTile pipe = (PipeTile) tileEntity;
+            Map<Direction, PipeUsableConfig> usableConfigs = input ? pipe.getConfig().getInputUsableConfig() : pipe.getConfig().getOutputUsableConfig();
+            if (usableConfigs.containsKey(pipePosition.getDirection())) {
+                ItemStack stack = usableConfigs.get(pipePosition.getDirection()).getFilter();
+                if (stack.isEmpty()) {
+                    return new Filter<ToTransport>() {};
+                }
+                if (stack.getItem() instanceof FilterItem) {
+                    return ((FilterItem)stack.getItem()).getFilter(stack);
+                }
+            }
+        }
+        LogManager.getLogger().error("Could not get Filter @" + pipePosition + " for " + (input ? "input" : "output"));
```

Comment:
Techarium.logger instead

Reply:
Wait, since when does this exist?
Will update it